kornia/geometry/nerf/nerf.py

```diff
@@ -0,0 +1,340 @@
+from kornia.utils import image_list_to_tensor
+
+
+class RayParameters:
+    def __init__(self):
+        self.NEAR, self.FAR = 0.0, 1.0  # ndc near far
+        self.N_SAMPLE = 128  # samples per ray
+        self.POS_ENC_FREQ = 10  # positional encoding freq for location
+        self.DIR_ENC_FREQ = 4  # positional encoding freq for direction
+
+
+class LearnFocals(nn.Module):
+    def __init__(self, h: int, w: int, num_cams: int, req_grad: bool, multi_focal: bool):
+        super().__init__()
+        self.__h = h
+        self.__w = w
+        self.__multi_focal = multi_focal
+        if multi_focal:
+            self.__fxfy = \
+                nn.Parameter(torch.tensor(torch.ones(size=(num_cams, 2)), dtype=torch.float32), requires_grad=req_grad)
+        else:
+            self.__fxfy = \
+                nn.Parameter(torch.tensor(torch.ones(size=(1, 2)), dtype=torch.float32), requires_grad=req_grad)
+
+    def forward(self, cam_id):
+        if self.__multi_focal:
+            fx = self.__fxfy[cam_id][0]
+            fy = self.__fxfy[cam_id][1]
+        else:
+            fx = self.__fxfy[0][0]
+            fy = self.__fxfy[0][1]
+        fxfy = torch.stack([fx**2 * self.__w, fy**2 * self.__h])
+        return fxfy
+
+    def is_multi_focal(self):
+        return self.__multi_focal
+
+
+class CameraCalibration:
+    def __init__(self,
+                 nerf_model: nn.Module = TinyNerf(pos_in_dims=63, dir_in_dims=27, D=128),
+                 device: str = 'cpu', checkpoint_path: str = None, checkpoint_save_frequency: int = 1,
+                 n_selected_rays: int = 32,
+                 lr: float = 0.001):
+        self._nerf_model: nn.Module = nerf_model
+        self._focal_net: nn.Module = None
+        self._pose_param_net: nn.Module = None
+
+        self._ray_params: RayParameters = None
+
+        self._opt_nerf: torch.optim.Optimizer = None
+        self._opt_focal: torch.optim.Optimizer = None
+        self._opt_pose: torch.optim.Optimizer = None
+
+        self._scheduler_nerf: MultiStepLR = None
+        self._scheduler_focal: MultiStepLR = None
+        self._scheduler_pose: MultiStepLR = None
+
+        self._device = device
+
+        self._checkpoint_path = checkpoint_path
+        self._checkpoint_save_frequency = checkpoint_save_frequency
+
+        self._n_selected_rays = n_selected_rays
+
+        self._lr = lr
+
+        self._multi_focal: bool = None
+
+        self._images: torch.Tensor = None
+        self._image_filenames: List[str] = None
+
+        self._epoch0 = 0
+
+    def load_images(self, path, num_of_images_to_use: int = None):
+        images: List[np.array] = []
+        self._image_filenames: List[str] = []
+        for i, filename in enumerate(os.listdir(path)):
+            if num_of_images_to_use is not None and i >= num_of_images_to_use:
+                break
+            img_bgr: np.array = cv2.imread(os.path.join(path, filename))
```

Comment:
possibly we should start really thinking to have a simple `kornia.io` to leverage image/video reading/writing. Based on recent survey seems that our community is an opencv fan. /cc @shijianjian @ducha-aiki @lferraz 
https://twitter.com/kornia_foss/status/1469232449917968386

Reply:
I modified the method `load_images`  to use `torchvision.io.read_image` to load all images in a directory, and then stack them to a 4D tensor. I removed the previous dependency on openCV and Numpy.

This function can be separated and moved to a Kornia module such as `kornia.io`